src/mlpack/tests/main_tests/linear_regression_test.cpp

```diff
@@ -18,49 +18,268 @@ static const std::string testName = "LinearRegression";
 
 using namespace mlpack;
 
-// Utility function to set a parameter and mark it as passed, using copy
-// semantics.
-template<typename T>
-void SetInputParam(const std::string& name, const T& value)
-{
-  CLI::GetParam<T>(name) = value;
-  CLI::SetPassed(name);
-}
-
-// Utility function to set a parameter and mark it as passed, using move
-// semantics.
+// Utility function to set a parameter and mark it as passed,
+// using copy semantics for lvalues and move semantics for rvalues
 template<typename T>
 void SetInputParam(const std::string& name, T&& value)
 {
-  CLI::GetParam<T>(name) = std::move(value);
+  CLI::GetParam<typename std::remove_reference<T>::type>(name)
+    = std::forward<T>(value);
   CLI::SetPassed(name);
 }
 
-struct LinearRegressionTestFixture
+struct LRTestFixture
 {
  public:
-  LinearRegressionTestFixture()
+  LRTestFixture()
   {
     // Cache in the options for this program.
     CLI::RestoreSettings(testName);
   }
 
-  ~LinearRegressionTestFixture()
+  ~LRTestFixture()
   {
     // Clear the settings.
     CLI::ClearSettings();
   }
 };
 
-BOOST_FIXTURE_TEST_SUITE(LinearRegressionMainTest, LinearRegressionTestFixture);
+void ResetSettings()
+{
+  CLI::ClearSettings();
+  CLI::RestoreSettings(testName);
+}
+
+BOOST_FIXTURE_TEST_SUITE(LinearRegressionMainTest, LRTestFixture);
+
+/**
+ * Training a model with different regularization parameter
+ * and ensuring that predictions are different
+ */
+BOOST_AUTO_TEST_CASE(LRDifferentLambdas)
+{
+  // required minimal difference between solutions
+  const double delta = 0.1;
+
+  arma::mat trainX = { 1.0, 2.0, 3.0 };
+  arma::mat testX = { 4.0 };
+  arma::rowvec trainY = { 1.0, 4.0, 9.0 };
```

Comment:
Looks like the version of Armadillo on AppVeyor cannot use initialization lists.  If you change this to the following it should fix the AppVeyor issue:

```
arma::mat trainX("1.0 2.0 3.0"); // This might need to be transposed after construction, not sure.
arma::mat testX("4.0");
arma::rowvec trainY("1.0 4.0 9.0");
```

Reply:
My code works if we use C++11 (I compiled it locally with Visual Studio), but it seems that Armadillo at AppVeyor build doesn't use

```
#define ARMA_USE_CXX11
```

Do have some agreement on this?

Reply:
I don't have a particular problem with forcing `ARMA_USE_CXX11` on Visual Studio, but I know that Conrad has had a lot of issues over the years with Visual Studio.  And VS C++11 support was pretty bad before the VS2015 release.  I would say, if it doesn't cause any problems, we could try it---after all, mlpack is using a lot of C++11 features even from the Visual Studio compiler.  mlpack already requires VS2015, so I *think* it will be ok.

I'm a bit concerned when you say that 7 tests are failing, and realized that we don't run the tests on AppVeyor to save time, since we have a limited time budget on AppVeyor.  I'll play with a PR to run the tests (#1188) and see what the results are.

Reply:
As for failing tests on Windows, I created an issue #1187 about memory leaks. I'll put there info about which tests are failing as well as about memory leaks

Reply:
Sounds good---we can handle the failing tests there.  In any case, if you want to force-enable `ARMA_USE_CXX11` on Windows I don't see a problem with it.  Probably it should just be added to `mlpack/prereqs.hpp` before Armadillo is included and that should solve the issue.

Reply:
I'm worried about the fact that I didn't find in Armadillo documentation any mention that compiled binary library doesn't depend on `ARMA_USE_CXX11`. 

However, when I tried, it worked fine and binary file seems the same (but I use an old version - 7.800.3). Nonetheless, I'm not sure that it's ok in general to compile Armadillo without this define and use with it because they could add something depending on `ARMA_USE_CXX11` in a cpp file in future releases. This could lead to some very hard memory issues.

My be the best solution is to merge this PR as it is (I removed using initializer_list constructors and now it compiles well on AppVeyor) and put a new task in todo list about enabling Armadillo's C++ features on AppVeyor. However, I think that it's important to enable it as described [here (search for ARMA_USE_CXX11)](http://arma.sourceforge.net/docs.html) and recompile

Reply:
Just realized that you are the developer of Armadillo. I guess you know this stuff very well :) I'll implement your suggestions about adding `ARMA_USE_CXX11` in mlpack/prereqs.hpp

Reply:
Ah, yeah, it looks like 6f213a3d07 actually added `ARMA_USE_CXX11` incorrectly.  Moving that up above the include of `arma_extend.hpp` should fix it just fine.

I think you edited the comment, but it seemed like a concern was that enabling `ARMA_USE_CXX11` would cause ABI incompatibilities for different versions of Armadillo.  The Armadillo wrapper library is only an optional wrapper around LAPACK/BLAS/other support libraries; its reason for existence is so that you can link with `-larmadillo` and not `-lblas -llapack -lother_stuff -lmore_other_stuff ...`.  So it doesn't actually have any Armadillo types in it, and I'm confident enough in the future development of Armadillo that `ARMA_USE_CXX11` won't ever affect the ABI.  I am a contributor to Armadillo, by the way, but it's Conrad who calls the shots. :)